### PR TITLE
switched from sequelize to typeorm

### PR DESCRIPTION
**Pull request by doodyparizada**

### CODE REVIEW

File: config/default.json [hunk 1]
```diff
@@ -6,19 +6,13 @@
...
-			"operatorsAliases": false,
-			"pool": {
-				"max": 5,
-				"min": 0,
-				"acquire": 30000,
-				"idle": 10000
-			}
-		}
+		"database": "database.sqlite",
+		"synchronize": true,
```
Comment (doodyparizada) [inline config/default.json]: I want `synchronize: false` and to have a separate runnable file that knows how to generate tables.

File: scripts/src/index.ts [hunk 2]
```diff
@@ -42,6 +47,9 @@ app.use((err, req, res, next) => {
 	res.status(status).send({ status, error: "Server error" });
 });
 
+// initializing db and models
+initModels().then(str => logger.debug(str));
```
Comment (doodyparizada) [inline scripts/src/index.ts]: msg?

File: scripts/src/models/index.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
+import "reflect-metadata";
+import { BaseEntity, createConnection } from "typeorm";
+import { ConnectionOptions } from "typeorm/connection/ConnectionOptions";
+
+import { getConfig } from "../config";
+import { normalizeError, path } from "../utils";
+
+const dbConfig = Object.assign(getConfig().db);
+if (dbConfig.type === "sqlite" && !/^[./]/.test(dbConfig.database)) {
```
Comment (doodyparizada) [inline scripts/src/models/index.ts]: what?

File: scripts/src/models/index.ts [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+		.then(connection => {
+			return createOnConnectedString(connection.options);
+		})
+		.catch(error => {
+			return normalizeError(error);
+		});
+}
+
+function createOnConnectedString(options: ConnectionOptions): string {
+	let str = `connected to ${ options.type } server`;
```
Comment (doodyparizada) [inline scripts/src/models/index.ts]: msg

File: scripts/src/models/offers.ts [hunk 5]
```diff
@@ -0,0 +1,185 @@
+import { Column, Entity, PrimaryColumn } from "typeorm";
+
+import { Model, Register } from "./index";
+
+export type OfferMeta = {
+	title: string;
+	image: string;
+	description: string;
+};
```
Comment (doodyparizada) [inline scripts/src/models/offers.ts]: 👍

File: scripts/src/models/offers.ts [hunk 6]
```diff
@@ -0,0 +1,185 @@
...
+	private _amount: number;
+
+	@Column({ name: "cap" })
+	private _cap: string;
+
+	@Column({ name: "meta" })
+	private _meta: string;
+
+	@Column({ name: "type" })
+	private _type: "spend" | "earn";
```
Comment (doodyparizada) [inline scripts/src/models/offers.ts]: this enum should be defined outside and used by all places that need it

File: scripts/src/models/offers.ts [hunk 7]
```diff
@@ -0,0 +1,185 @@
...
+	public get createdDate(): Date {
+		return this._createdDate;
+	}
+
+	public get type(): "coupon" {
+		return this._type;
+	}
+
+	public get isUsed(): boolean {
+		return this._isUsed;
```
Comment (doodyparizada) [inline scripts/src/models/offers.ts]: I'm thinking that we may want usedBy

File: scripts/src/models/users.ts [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+	private _appId: string;
+
+	@Column({ name: "app_user_id" })
+	private _appUserId: string;
+
+	@Column({ name: "created_date" })
+	private _createdDate: Date;
+
+	@Column({ name: "activated_date" })
+	private _activatedDate: Date;
```
Comment (doodyparizada) [inline scripts/src/models/users.ts]: reminder: yohay suggested to have a permissions object that each token encapsulates and a user has to allow permissions per token.

File: scripts/src/models/users.ts [hunk 9]
```diff
@@ -0,0 +1,139 @@
...
+
+	@Column({ name: "name" })
+	private _name: string;
+
+	@Column({ name: "jwt_public_key" })
+	private _jwtPublicKey: string;
+
+	constructor() {
+		super();
+	}
```
Comment (doodyparizada) [inline scripts/src/models/users.ts]: Do you have to define a constructor if all it does is call super?

File: config/default.json (hunk 1 above)
Comment (nitzantomer) [reply config/default.json]: This is part of the `default` config, which (at least for now) is the "dev config".
In a different config file you can set it to `false`, but for development it is easier with it as `true`

File: scripts/src/models/index.ts (hunk 3 above)
Comment (nitzantomer) [reply scripts/src/models/index.ts]: Well, if using sqlite and the db file path isn't relative/absolute then make the path relative to the root directory of the project.
For example, take what's in our `config/default.json` configuration:
```
...
"database": "database.sqlite",
...
```
The file that will be used will be `PROJECT_ROOT/database.sqlite`

File: scripts/src/models/offers.ts (hunk 7 above)
Comment (nitzantomer) [reply scripts/src/models/offers.ts]: Maybe, so we'll add it when/if needed

**Comment by doodyparizada** [comment]
I rebased from master. The main conflict is that I added default values to the models which we need to reimplement according to b68153d13485a082638804ee33336b1f34bb08ef

**Pull request merged**